backend/dataall/modules/shares_base/db/share_object_repositories.py

```
@@ -194,6 +201,16 @@ def get_share_data_items_by_type(session, share, share_type_model, share_type_ur
             query = query.filter(ShareObjectItem.healthStatus == healthStatus)
         return query.all()
 
+    @staticmethod
+    def get_share_object_with_health_status(session, health_status_list: List[str] = None):
+        query = (
+            session.query(ShareObject)
+            .join(ShareObjectItem, ShareObjectItem.shareUri == ShareObject.shareUri)
+            .filter(ShareObjectItem.healthStatus.in_(health_status_list))
```

won't that break if we call it with `health_status_list=None`? isn't it better to pass an empty list?

---

You are right. Will make the correction. It will certainly return incorrect results if not break

---

Corrected this. Thanks for pointing this out

---

Removed the default value on the input argument argument. Will let it fail or return empty list is None is provided